web/src/features/datasets/components/DatasetRunItemsTable.tsx

```diff
@@ -34,6 +38,9 @@ type RowData = {
   scores: APIScore[];
   latency?: number;
   totalCost?: string;
+
+  // any number of additional detail columns for individual scores
+  [key: string]: any; // any of type APIScore[] for detail columns
```

Comment:
is there a way around this to still get type errors on other columns?

Reply:
Unfortunately not really. Typescript does not support regex on strings, which we would need here to type the string more explicitly and inform typescript that there can not be any overlap between the dynamic keys and the predefined ones. I would have preferred to avoid this implicit technical complexity, but could not find a robust way to achieve this